pkg/jobs/jobsprofiler/profiler.go

```diff
@@ -46,6 +55,22 @@ func StorePlanDiagram(
 
 			dspKey := profilerconstants.MakeDSPDiagramInfoKey(timeutil.Now().UnixNano())
 			infoStorage := jobs.InfoStorageForJob(txn, jobID)
+
+			// Limit total retained diagrams by culling older ones as needed.
+			count, err := infoStorage.Count(ctx, profilerconstants.DSPDiagramInfoKeyPrefix, profilerconstants.DSPDiagramInfoKeyMax)
+			if err != nil {
+				return err
+			}
+			const keep = MaxRetainedDSPDiagramsPerJob - 1
```

Comment:
to discuss: we should consider pushing this logic into `infoStorage.Write()` to prevent any info storage user from bloating the job system.

Reply:
I'm not so sure about that: info storage is generic, and Write doesn't know what rows are critical to a job's correctness and which are able to be culled when they're not needed.

Furthermore, the _real_ answer here, I think, is that we just should never be scanning job_info in the first place (except to delete a job or make a debug.zip); the whole point of this table was that we could have large and/or slow writers to some keys in this table without affecting "the job system" but clearly we did not realize that in the current access patterns since we never finished the transition.